crates/supervisor/core/src/supervisor.rs

```diff
@@ -61,6 +67,15 @@ pub trait SupervisorService: Debug + Send + Sync {
     /// Returns the finalized L1 block that the supervisor is synced to.
     fn finalized_l1(&self) -> Result<BlockInfo, SupervisorError>;
 
+    /// Returns the [`SuperRootResponse`] at a specified timestamp, which represents the global
+    /// state across all monitored chains.
+    ///
+    /// [`SuperRootResponse`]: kona_interop::SuperRootResponse
+    async fn super_root_at_timestamp(
+        &self,
+        timestamp: u64,
+    ) -> Result<SuperRootResponse, SupervisorError>;
```

Comment:
it seems counterintuitive that the rpc response is assembled in the service. it would be more adequate if this returned only a tuple of super roots and chain infos. that way, the service handles processing which requires communication with db or managed node, but nothing more. the rpc server then handles only communication with the service, and any further processing of the data it returns into the rpc response format.